article.md

```diff
@@ -6,16 +6,38 @@ Une nouvelle version du langage [Python](https://www.python.org/) (et par consé
 
 Les plus pressés peuvent profiter de ce court résumé [des principales nouveautés](https://docs.python.org/3.5/whatsnew/3.5.html) :
 
- - [PEP 492](https://www.python.org/dev/peps/pep-0492) : les coroutines deviennent une construction spécifique du langage. Cette gestion dans l'interpréteur se fait via deux nouveaux mots-clés (`async` et `await`) et vise à compléter le support de la « programmation asynchrone » dans Python.
- - [PEP 465](http://www.python.org/dev/peps/pep-0465) : l'opérateur binaire `@` est introduit pour gérer la multiplication matricielle.
- - [PEP 484](https://www.python.org/dev/peps/pep-0484/) : les annotations apposables sur les paramètres et la valeur de retour des fonctions et méthodes sont maintenant standardisées et servent uniquement à préciser le type de ces éléments. Les annotations ne sont toujours pas utilisées par l'interpréteur et cette PEP n'est constituée que de conventions.
- - [PEP 448](https://www.python.org/dev/peps/pep-0448/) : les opérations d'*unpacking* sont généralisées et permettent maintenant d'être combinées.
-
-<--COMMENT
-
-Pour les 4 principaux thèmes, une section avec le contexte, ce que ça apporte et une note de conclusion
-
-COMMENT-->
+ - [PEP 492](https://www.python.org/dev/peps/pep-0492) : les coroutines deviennent une construction spécifique du langage. Cette gestion dans l'interpréteur se fait via deux nouveaux mots-clés (`async` et `await`) et vise à compléter le support de la « programmation asynchrone » dans Python. Ces deux mot-clés permettent d'écrire des coroutines utilisables avec `asyncio` de façon similaire à des fonctions python classique. Par exemple :
```

Comment:
Oui. Les deux sont équivalent et peuvent être utilisé l'un à la place de
l'autre partout (pour le moment). Mais il n'est pas impossible que des
différences soient faites à l'avenir. Du coup ces mots clés ont été rajouté
: si des ajouts doivent être fait, ce sera uniquement pour les nouvelles
coroutine.

D'ailleurs `async for` et `async with` ne peuvent d'utiliser que dans des
coroutine définies avec `async` et pas dans des générateurs.

Christophe
Le 24 août 2015 20:01, "Vincent" <notifications@github.com> a écrit :

> In article.md
> <https://github.com/Vayel/Python-3.5/pull/10#discussion_r37780693>:
>
> > @@ -6,16 +6,38 @@ Une nouvelle version du langage [Python](https://www.python.org/) (et par consé
> >
> >  Les plus pressés peuvent profiter de ce court résumé [des principales nouveautés](https://docs.python.org/3.5/whatsnew/3.5.html) :
> >
> > - - [PEP 492](https://www.python.org/dev/peps/pep-0492) : les coroutines deviennent une construction spécifique du langage. Cette gestion dans l'interpréteur se fait via deux nouveaux mots-clés (`async` et `await`) et vise à compléter le support de la « programmation asynchrone » dans Python.
> > - - [PEP 465](http://www.python.org/dev/peps/pep-0465) : l'opérateur binaire `@` est introduit pour gérer la multiplication matricielle.
> > - - [PEP 484](https://www.python.org/dev/peps/pep-0484/) : les annotations apposables sur les paramètres et la valeur de retour des fonctions et méthodes sont maintenant standardisées et servent uniquement à préciser le type de ces éléments. Les annotations ne sont toujours pas utilisées par l'interpréteur et cette PEP n'est constituée que de conventions.
> > - - [PEP 448](https://www.python.org/dev/peps/pep-0448/) : les opérations d'*unpacking* sont généralisées et permettent maintenant d'être combinées.
> > -
> > -<--COMMENT
> > -
> > -Pour les 4 principaux thèmes, une section avec le contexte, ce que ça apporte et une note de conclusion
> > -
> > -COMMENT-->
> > + - [PEP 492](https://www.python.org/dev/peps/pep-0492) : les coroutines deviennent une construction spécifique du langage. Cette gestion dans l'interpréteur se fait via deux nouveaux mots-clés (`async` et `await`) et vise à compléter le support de la « programmation asynchrone » dans Python. Ces deux mot-clés permettent d'écrire des coroutines utilisables avec `asyncio` de façon similaire à des fonctions python classique. Par exemple :
>
> En fait je crois que mon incompréhension vient de mon ignorance vis-à-vis
> de la l'implémentation de la prog asynchrone en Python. Je pensais qu'une
> coroutine avec async et await était la même chose qu'un générateur, utilisé
> jusque-là ; ces derniers n'étant pas rattachés à une lib mais natifs au
> langage, "Ces coroutines sont avant tout faites pour être utilisé avec
> asyncio." me paraît bizarre. Mais je crois que j'ignore en fait ce que font
> async et await.
>
> async remplace le décorateur et await le yield from ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Vayel/Python-3.5/pull/10/files#r37780693>.
>